### ISSUE TITLE
don't cache psi elements, use com.intellij.psi.util.CachedValue

### ISSUE THREAD

**Issue by shalom938** [digma-ai/digma-intellij-plugin#250]
is it not advisable to cache psi elements because the psi tree may be rebuilt while the IDE is running and cached psi element may become obsolete.
a recommended way to cache psi elements is to use com.intellij.psi.util.CachedValue.
https://plugins.jetbrains.com/docs/intellij/psi-performance.html#cache-results-of-heavy-computations

Currently DocumentInfoContainer caches the PsiFile reference.
some of the endpoint frameworks like JaxrsFramework, MicronautFramework, GrpcFramework also cache psi elements.

**Comment by arik-dig** [comment]
I'm caching the basic attribute, to determine for example, if Micronaut is available at all or not, 
same for JAX-RS and GRPC, otherwise, searching the entire Project + 3rd parties each time we open a file is quite heavy, isn't it?

**Comment by shalom938** [comment]
You're right, but com.intellij.psi.util.CachedValue is meant to help with that, see here
https://plugins.jetbrains.com/docs/intellij/psi-performance.html#cache-results-of-heavy-computations

**Comment by arik-dig** [comment]
@shalom938 I started looking at it, its very unclear what should we do in order to make the cache dirty, so it will re-evaluate the cached value, are you familiar with it?
